pkgs/development/tools/jq/default.nix

```diff
@@ -1,44 +1,98 @@
-{ stdenv, nixosTests, fetchurl, oniguruma }:
+{ stdenv, buildPackages, bundlerUpdateScript, fetchFromGitHub, nixosTests
+, oniguruma
+# installCheck inputs
+, bundlerApp ? null
+# features
+# (enabling Valgrind checks for testing severly increases build time)
+, enableValgrindChecks ? false
+, less ? null, valgrind ? null, which ? null
+}:
+
+assert enableValgrindChecks -> valgrind != null;
+assert enableValgrindChecks -> which != null;
```

Comment:
Maybe just have `less`/`valgrind`/`which` in the arguments without `? null` and drop this assertion? They won't be used anyway except conditionally (and evaluation is lazy anyway).

Reply:
Isn't this pattern meant for `callPackage` on the file's expression from scopes that don't necessarily have optional package arguments present, though? Destructuring attrset arguments eagerly check for their named parameters. Admittedly, it's a niche case, but without an overall style decision in Nixpkgs I'd prefer to keep with the usual style (that has minor technical merits).

The assertions are there for clearer error messages, and are a pattern (to some extent) throughout Nixpkgs too AFAIK.

Thanks for reminding me to take a look at `less`, not sure actually why that's there at the moment.

Reply:
Think I included `less` for `$src/tests/mantest`, but it runs fine without it. Removed.

Reply:
> I'd prefer to keep with the usual style

Usually style is to add an enable flag and use lib.optionals.